implementations/rust/ockam/ockam/src/pipe2/mod.rs

```diff
@@ -0,0 +1,107 @@
+// If `pipe` is so great, why is there no `pipe2`?
```

Comment:
... Well, it's sure hard to argue with that 😂. Surprised we don't get this from users, even.

(TBH, I do think pipe isn't an ideal name for either of them, since it's so close and easily confused with channel, but that for sure doesn't need to be solved here)

Reply:
Yea, the terminology of "pipe", "channel", and "stream" is a bit overlapping and confusing. I'd be especially curious to hear from unix developers who have a very different understanding of what a pipe is as to how this lingo may be confusing.

It could be something we'll just have to accept and have good documentation (and visualisations) on and then we need to use those terms consistently.